crates/wysiwyg/src/dom/dom_struct.rs

```diff
@@ -176,6 +178,22 @@ where
         find_range::find_range(self, start, end)
     }
 
+    pub fn find_range_by_node(&self, node_handle: &DomHandle) -> Range {
+        let result = find_range::find_pos(self, node_handle, 0, usize::MAX);
+
+        let locations = match result {
+            FindResult::Found(locations) => locations,
+            _ => panic!("Node does not exist"),
+        };
+
+        let leaves = locations.iter().filter(|l| l.is_leaf());
+
+        let s = leaves.clone().map(|l| l.position).min().unwrap();
+        let e = leaves.clone().map(|l| l.position + l.length).max().unwrap();
```

Comment:
You probably don't need a 2nd clone here?

Reply:
[`a753a13`](https://github.com/matrix-org/matrix-rich-text-editor/pull/423/commits/a753a134040615916b28415de7c68bb257e9d686)